### PR TITLE
Add custom order

### PR DESCRIPTION
**Pull request by SavaCool122**
Close: #39

### CODE REVIEW

**Comment by SavaCool122** [comment]
@Siilwyn 🏓 
Could you take a look at this PR?

File: src/main.mjs [hunk 1]
```diff
@@ -62,6 +71,14 @@ export default {
...
       default: true,
     },
+    propertiesOrder: {
+      type: "string",
+      array: true,
+      description:
+        "An array of declaration names to sort according to their index in the array.",
+      category: "css-declaration-sorter",
+      default: [{ value: [] }],
+    },
```
Comment (Siilwyn) [inline src/main.mjs:81]: Could passing an array be one of the choices in `cssDeclarationSorterOrder`? Or is that invalid/unsupported by Prettier?

Comment (SavaCool122) [reply src/main.mjs:81]: As i can see here support any type:
https://github.com/prettier/prettier/blob/d970deed6b6cabe21fdb35249265d766932b5718/src/index.d.ts#L753-L762
But at the moment of using the normalizer inside prettier shows that the parameters are invalid. Maybe I'm missing something, but that's what I see in the source code.

Comment (Siilwyn) [reply src/main.mjs]: I see it now, sadly Prettier doesn't support arbitrary values as a choice, ah well. :)

**Comment by Siilwyn** [comment]
@SavaCool122 made a couple of changes:
- I've renamed the option so it's prefixed with `cssDeclarationSorter`, as [discussed here](https://github.com/Siilwyn/prettier-plugin-css-order/issues/22). Let me know what you think of the `...CustomOrder` name.
- Inlined the `createSorter` function as we only use it once, simplifies the code a bit.
- Renamed `findSorter` to `resolveSorterOption`, do you think this clarifies the function's purpose?
- Changed the option description
- Changed the tests so they don't pass if the new option is unset.
- :warning: Removed the `default` empty array value, I don't think having a default value for the custom order makes sense. However I have no idea if removing this comes with any drawbacks, please let me know if it does!

**Comment by SavaCool122** [comment]
First of all, thanks for this suggestion and fixes!

> I've renamed the option so it's prefixed with cssDeclarationSorter, as https://github.com/Siilwyn/prettier-plugin-css-order/issues/22. Let me know what you think of the ...CustomOrder name.

LGTM

> Renamed findSorter to resolveSorterOption, do you think this clarifies the function's purpose?

It looks better than mine, I like it

>  Changed the tests so they don't pass if the new option is unset.

Thanks for test fixes. I appreciate it.

> ⚠️ Removed the default empty array value, I don't think having a default value for the custom order makes sense. However I have no idea if removing this comes with any drawbacks, please let me know if it does!

I think it looks clearer without the default value, great suggestion!

**Comment by SavaCool122** [comment]
We're both happy with the current implementation. Can we merge it now?

**Review by Siilwyn — APPROVED** [review]

**Comment by Siilwyn** [comment]
Published! Thank you for your contributions!

**Pull request merged**